packages/devtools_app/integration_test/test/live_connection/eval_and_browse_test.dart

```diff
@@ -116,5 +124,82 @@ class _EvalTester {
     await tester.pumpAndSettle();
     await simulateKeyDownEvent(LogicalKeyboardKey.enter);
     await simulateKeyUpEvent(LogicalKeyboardKey.enter);
+    await tester.pump(longPumpDuration);
+  }
+
+  Future<void> switchToSnapshotsAndTakeOne() async {
+    // Open memory screen.
+    await switchToScreen(tester, ScreenMetaData.memory);
+
+    // Close warning and chart to get screen.
+    await tapAndPump(find.byKey(DebugModeMemoryMessage.closeKey));
+    await tapAndPump(find.text(PrimaryControls.memoryChartText));
+
+    // Make console wider.
+    // The distance is big enough to see more items in console,
+    // but not too big to make classes in snapshot hidden.
+    const dragDistance = -320.0;
+    await tester.drag(
+      find.byKey(ConsolePaneHeader.theKey),
+      const Offset(0, dragDistance),
+    );
+    await tester.pumpAndSettle();
+
+    // Switch to diff tab.
+    await tapAndPump(find.text('Diff Snapshots'));
+
+    logStatus('Started taking snapshot.');
+    // Take snapshot.
+    const snapshotDuration = Duration(seconds: 20);
+    await tapAndPump(
+      find.byIcon(iconToTakeSnapshot),
+      duration: snapshotDuration,
+    );
+    logStatus('Finished taking snapshot.');
+
+    // Sort by class.
+    await tapAndPump(find.text('Class'));
+
+    // Select class.
+    await tapAndPump(find.text('MyApp'));
+  }
+
+  /// Taps and settles.
+  ///
+  /// If [next] is provided, will repeat the tap till [next] combined with [at] returns results.
+  /// If [next] is not null returns [next]  combined with [at, otherwise returns [finder].
+  Future<Widget?> tapAndPump(
+    Finder finder, {
+    Duration? duration,
+    Finder? next,
+  }) async {
+    int tryNumber = 0;
+
+    Future<void> action() async {
+      logStatus('tapping #$tryNumber to find \n'
+          '[$finder]\n');
+      tryNumber++;
+      await tester.tap(finder);
+      await tester.pump(duration);
+      await tester.pumpAndSettle();
```

Comment:
we shouldn't need both pump and pumpAndSettle

Reply:
pumpAndSettle does not take nullable duration, so code is simpler this way

Reply:
pumpAndSettle does take a duration parameter. if you pass the duration there, then you can remove the call to pump.

Reply:
duration here is nullable
pumpAndSettle takes non-nullable duration

Yes, it can be handled in the code at cost of readability. I prefer to stay with readability.